mailgun/decorators.py

```diff
@@ -0,0 +1,31 @@
+import hashlib
+import hmac
+
+from functools import wraps
+
+from django.conf import settings
+from django.utils.decorators import available_attrs
+from django.shortcuts import redirect
+from django.core.urlresolvers import reverse_lazy
+
+
+def authenticate(redirect_url=reverse_lazy('mailgun:auth_error'), raise_exception=False):
+    def decorator(view_func):
+        @wraps(view_func, assigned=available_attrs(view_func))
+        def _wrapped_view(request, *args, **kwargs):
+            timestamp = request.POST.get('timestamp')
```

Comment:
Not sure what the spec for the signature is, but typically you want to refuse to verify if the timestamp is outside a certain window.